src/components/migrated-bch.js

```diff
@@ -0,0 +1,324 @@
+    const bchWallets = await bch.wallets().list();
+    const migratedWallet = _.find(bchWallets.wallets, w => w._wallet.migratedFrom === this.state.walletId);
+
+    if (!migratedWallet) {
+      throw new Error('could not find a bch wallet which was migrated from ' + this.state.walletId);
+    }
+
+    console.info('found bch wallet: ', migratedWallet.id());
+
+    let recoveryTx;
+    try {
+      recoveryTx = await this.createRecoveryTx(bch, migratedWallet);
+    } catch (e) {
+      if (e.message === 'insufficient balance') { // this is terribly unhelpful
+        e.message = 'Insufficient balance to recover';
+      }
+      this.collectLog(e.message);
+      this.setState({ error: e.message, recovering: false });
+    }
+
+    if (!recoveryTx || !recoveryTx.hex) {
+      console.error('Failed to create half-signed recovery transaction');
+      return;
+    }
+
+    let needsUnlock = false;
+    try {
+      await migratedWallet.submitTransaction({
+        txHex: recoveryTx.hex
+      });
+    } catch (e) {
+      if (e.message === 'needs unlock') {
+        // try again after unlocking
+        needsUnlock = true;
+      } else {
+        throw e;
+      }
+    }
+
+    if (needsUnlock) {
+      try {
+        await bitgo.unlock({ otp: this.state.twofa });
+        recoveryTx = await migratedWallet.submitTransaction({
+          txHex: recoveryTx.hex
+        });
+      } catch (e) {
+        // failed even after unlock - this is fatal
+        console.log('got error on submit after unlock');
+        console.error(e);
+        throw e;
+      }
+    }
+    this.setState({ recoveryTx, recovering: false });
+  };
+
+  saveTransaction = () => {
+    const fileData = this.state.recoveryTx;
+    let fileName;
+    const filePrefix = this.props.bitgo.env === 'prod' ? 'bch' : 'tbch';
+
+    fileName = `${filePrefix}r-${moment().format('YYYYMMDD')}.signed.json`;
+
+    const dialogParams = {
+      filters: [{
+        name: 'Custom File Type',
+        extensions: ['json']
+      }],
+      defaultPath: '~/' + fileName
+    };
+
+    // Retrieve the desired file path and file name
+    const filePath = dialog.showSaveDialog(dialogParams);
+    if (!filePath) {
+      // TODO: The user exited the file creation process. What do we do?
+      return;
+    }
+
+    try {
+      fs.writeFileSync(filePath, JSON.stringify(fileData, null, 4), 'utf8');
+    } catch (err) {
+      console.log('error saving', err);
+      this.setState({ error: 'There was a problem saving your recovery file. Please try again.' });
+    }
+  };
+
+  render() {
+    const coin = this.props.bitgo.env === 'prod' ? 'bch' : 'tbch';
+
+    return (
+      <div>
+        <h1 className='content-header'>Migrated Bitcoin Cash Recoveries</h1>
+        <p className='subtitle'>This tool will help you recover Bitcoin Cash from migrated wallets which are no longer officially supported by BitGo.</p>
+        <Alert color='warning'>
+          Do we need a warning here?
```

Comment:
Not if we don't have anything to warn about :)

Reply:
Will remove.